swan-lake/learn-the-platform/ballerina-persist/persist-client-api.md

```diff
@@ -0,0 +1,245 @@
+---
+layout: ballerina-persist-documentation-left-nav-pages-swanlake
+title: Persist Client API
+description: The sections gives details on the generated Persist Client API.
+keywords: ballerina, programming language, ballerina packages, persist, persist client, persist client api
+permalink: /learn/ballerina-persist/persist-client-api/
+active: persist_client_api
+intro: Persist client is used to perform CRUD operations on the data source. Each entity type will have five resource methods for each operation. The resource methods are get, get(get by identity), post, put, and delete. Persist CLI tool will generate the derived Entity Types and the Clients from the model definition.
+redirect_from:
+- /learn/ballerina-persist/persist-client-api
+---
+
+## Derived Entity Types
+
+Ballerina record types used in the generated clients are derived from the entity types defined in the data model.
+
+Example of a defined entity,
+```ballerina
+type Workspace record {|
+    readonly string id;
+    string workspaceType;
+    Building location; // 1-n relation (parent)
+    Employee? employees; // 1-1 relation (child)
+|};
+```
+
+There are six types of derived entity types:
+1. Entity Types  
```

Comment:
```suggestion
### `Entity` types  
```
Shall we add the types as sub headings? Can give a full list and then ad sub-sections for each as we have a considerable list of 6.